server/handler/contributor.go

```diff
@@ -0,0 +1,273 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+	"gorm.io/gorm"
+
+	"github.com/samouraiworld/topofgnomes/server/onchain"
+)
+
+type githubUserResponse struct {
+	ID                 string           `json:"id"`
+	Login              string           `json:"login"`
+	AvatarUrl          string           `json:"avatarUrl"`
+	URL                string           `json:"url"`
+	Name               string           `json:"name"`
+	Bio                string           `json:"bio"`
+	Location           string           `json:"location"`
+	JoinDate           string           `json:"joinDate"`
+	WebsiteUrl         string           `json:"websiteUrl"`
+	TwitterUsername    string           `json:"twitterUsername"`
+	TotalStars         int              `json:"totalStars"`
+	TotalRepos         int              `json:"totalRepos"`
+	Followers          int              `json:"followers"`
+	Following          int              `json:"following"`
+	TotalCommits       int              `json:"totalCommits"`
+	TotalPullRequests  int              `json:"totalPullRequests"`
+	TotalIssues        int              `json:"totalIssues"`
+	RecentIssues       []recentActivity `json:"recentIssues"`
+	RecentPullRequests []recentActivity `json:"recentPullRequests"`
+	TopRepositories    []topRepository  `json:"topRepositories"`
+	Wallet             string           `json:"wallet"`
+	GnoBalance         string           `json:"gnoBalance"`
+}
+
+type topRepository struct {
+	NameWithOwner   string `json:"nameWithOwner"`
+	Description     string `json:"description"`
+	URL             string `json:"url"`
+	StargazerCount  int    `json:"stargazerCount"`
+	PrimaryLanguage string `json:"primaryLanguage"`
+}
+
+type recentActivity struct {
+	Title      string `json:"title"`
+	URL        string `json:"url"`
+	CreatedAt  string `json:"createdAt"`
+	Repository string `json:"repository"`
+	Type       string `json:"type"`
+}
+
+type cachedUser struct {
+	Data      githubUserResponse
+	Timestamp time.Time
+}
+
+var (
+	userCache   = make(map[string]cachedUser)
+	cacheMutex  sync.Mutex
```

Comment:
do you have a preference for this implementation ?
we're using ristretto on the other APIs